%YAML:1.2
RealSenseColorCamera:
  - name: "realsense_color"

RealSenseDepthCamera:
  - name: "realsense_depth"

NormalColorViewer:
  - name: "color_viewer"
    color_camera: "realsense_color"
    renderer_geometry: "renderer_geometry"

NormalDepthViewer:
  - name: "depth_viewer"
    depth_camera: "realsense_depth"
    renderer_geometry: "renderer_geometry"

RendererGeometry:
  - name: "renderer_geometry"
    bodies: ["cube"]

Body:
  - name: "cube1"
    metafile_path: "./cube1.yaml"

# Body:
#   - name: "cube2"
#     metafile_path: "./cube2.yaml"

ManualDetector:
  - name: "cube_manual_detector"
    metafile_path: "./cube_manual_detector.yaml"
    body: "cube"
    color_camera: "realsense_color"

RegionModel:
  - name: "cube_region_model"
    metafile_path: "./default_model.yaml"
    body: "cube"

DepthModel:
  - name: "cube_depth_model"
    metafile_path: "./default_model.yaml"
    body: "cube"

RegionModality:
  - name: "cube_region_modality"
    body: "cube"
    color_camera: "realsense_color"
    region_model: "cube_region_model"
    measure_occlusions: {depth_camera: "realsense_depth"}

DepthModality:
  - name: "cube_depth_modality"
    body: "cube"
    depth_camera: "realsense_depth"
    depth_model: "cube_depth_model"

Optimizer:
  - name: "cube_optimizer"
    modalities: ["cube_region_modality", "cube_depth_modality"]

Tracker:
  - name: "tracker"
    viewers: ["color_viewer", "depth_viewer"]
    detectors: ["cube_manual_detector"]
    optimizers: ["cube_optimizer"]